core/remote/watcher.js

```diff
@@ -285,15 +285,18 @@ class RemoteWatcher {
             // move inside move
             change.was.path = remoteChange.applyMoveToPath(previousChange, change.was.path)
             change.needRefetch = true
-            return change
+            return change // FileMove
           } else {
-            return {
+            const descendantChange = {
               sideName,
-              type: 'IgnoredChange',
+              type: 'DescendantChange',
+              update: change.update,
               doc,
               was,
-              detail: `File was moved as descendant of ${_.get(previousChange, 'doc.path')}`
+              ancestorPath: _.get(previousChange, 'doc.path')
             }
+            if (previousChange.type === 'DirMove') remoteChange.includeDescendant(previousChange, descendantChange)
```

Comment:
Isn't the `if` check useless? (since already true from line 283)

Reply:
Flow :sparkles: